codalab/worker/bundle_state.py

```diff
@@ -33,19 +36,102 @@ class State(object):
     FINAL_STATES = {READY, FAILED, KILLED}
 
 
+DependencyKey = namedtuple('DependencyKey', 'parent_uuid parent_path')
```

Comment:
This almost feels like it should be generic to CodaLab, not just the worker...  Maybe with the exception of `location`...maybe should say what this is...

Reply:
Will add comments explaining for now. I agree that there's more room for canonicalizing these across Codalab, but I don't have the bandwidth to undertake it all at the moment, and once the object definitions are in, it is less work/breaking changes to move them around and use them in more places.